packages/react-core/src/components/Tabs/Tabs.tsx

```diff
@@ -18,9 +18,12 @@ export enum TabsComponent {
   nav = 'nav'
 }
 
+type ChildElement = React.ReactElement<TabProps, React.JSXElementConstructor<TabProps>>;
+type Child = ChildElement | boolean | null | undefined;
+
 export interface TabsProps extends Omit<React.HTMLProps<HTMLElement | HTMLDivElement>, 'onSelect'>, OUIAProps {
-  /** Content rendered inside the tabs component. Must be React.ReactElement<TabProps>[] */
-  children: React.ReactNode;
+  /** Content rendered inside the tabs component. */
+  children: Child | Child[];
```

Comment:
Technically this is a breaking change.  It probably would not impact anyone but we try and not introduce anything that could be breaking like more restrictive type changes.  We can pull this update into our next breaking change release.

Reply:
Based on the issue - it sounds like this type enforcement is already necessary - using any other type would already produce a broken tab

Reply:
Although the type change is technically breaking, the runtime behavior is actually improved. Before if you'd pass in a [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) value the component would attempt to render it, and since there are no props associated with these children this will cause a run-time error (see [this example](https://codesandbox.io/s/practical-elbakyan-b99oj)).

This is caused by the fact that before the children were filtered using a [boolean filter](https://github.com/patternfly/patternfly-react/pull/6767/files#diff-b19cd68b4e0b4ab51ece072430fafe67e2a064f4241b44a923867085f04f6497L312), which would allow truthy values that are not React elements. In this PR these filters are replaced by [`isValidElement()`](https://reactjs.org/docs/react-api.html#isvalidelement), which asserts the value passed is a valid React element created with [`createElement()`](https://reactjs.org/docs/react-api.html#createelement), meaning the required props will be present.

So considering that there could realistically not be any applications out there doing this, unless they created a new component that matches all of the required props of `Tab` I don't see this as a breaking change.